collection/spec/collection.go

```diff
@@ -343,66 +343,75 @@ func AssertCollectionEventListenerSpec(t *testing.T, constructor Constructor) {
 			t.Fatalf("listing without providing any keyIDs should error: %q", err)
 		}
 
+		scopedCtx := profile.AddIDToContext(ctx, missPiggy.ID.Encode())
```

Comment:
A recommendation for doing this particular piece differently: if you look at the test user_3_pull_dataset, `scopedCtx` gets constructed inside a plain `{ ... }` block (no explicit control flow) and then immediately is used, rather than being kept as a local var that gets reused in the rest of the test. This accomplishes two subtle things: (1) better simulates the way this object is used in real code and (2) really makes references to `scopedCtx` stand out. There's no need to carefully read other parts of the code to see whether or not `ctx` or `scopedCtx` is being used for each call, because the plain block guarantees that the scope is not visible outside of it.